lib/elixir/lib/module/types/apply.ex

```diff
@@ -566,11 +566,21 @@ defmodule Module.Types.Apply do
     is_map_key(builtin_modules(), module)
   end
 
+  @builtin_protocols [
+    Collectable,
+    Enumerable,
+    IEx.Info,
+    Inspect,
+    JSON.Encoder,
+    List.Chars,
+    String.Chars
+  ]
+
   defp builtin_modules do
     case :persistent_term.get(__MODULE__, nil) do
       nil ->
         {:ok, mods} = :application.get_key(:elixir, :modules)
-        mods = Map.from_keys(mods, [])
+        mods = Map.from_keys(mods -- @builtin_protocols, [])
```

Comment:
`IEx.Info` is not in `:elixir` app so this filtering will not work as intended. `:iex` app is not loaded in most cases. Does it need special handing?

Reply:
Good catch. In this case, since we want to discard it, it is fine. It also keeps us safe if we do chose to load IEx in the future. :)